lib/stistools/inttag.py

```diff
@@ -1,5 +1,364 @@
+    # Mask events in imset if there are any lapses in GTI
+    gti_mask = np.array([False] * len(imset_events))  # Start by assuming all events are outside all GTIs
+    for gti in gti_data:
+        # Create mask of events within GTI
+        mask = (imset_events['TIME'] > gti[0]) * (imset_events['TIME'] < gti[1])
+        gti_mask = np.logical_or(gti_mask, mask)  # OR global gti mask with local gti mask
+
+    good_events = imset_events[gti_mask]  # All events in the imset within the GTI(s)
+    if len(good_events) == 0:
+        exp_time = 0
+        expstart = tzero_mjd
+        expstop = tzero_mjd
+        return exp_time, expstart, expstop, good_events
+    expstart = tzero_mjd + good_events['TIME'][0] / sec_per_day  # exposure start in MJD for imset
+    expstop = tzero_mjd + good_events['TIME'][-1] / sec_per_day  # exposure stop in MJD for imset
+
+    # Determine GTI gap regions
+    gaps = []
+    if len(gti_data) > 1:
+        for i, gti in enumerate(gti_data):
+            if i == 0:
+                continue
+            gaps.append((gti_data[i - 1][1], gti[0]))
+
+    # Calculate exposure time lost due to non-GTI overlap
+    exptime_loss = 0
+    for gap in gaps:
+        if gap[1] <= stoptime and gap[0] >= starttime:
+            exptime_loss += gap[1] - gap[0]
+        elif gap[1] >= stoptime and gap[0] <= stoptime:
+            exptime_loss += stoptime - gap[0]
+        elif gap[1] >= starttime and gap[0] <= starttime:
+            exptime_loss += gap[1] - starttime
+        else:
+            continue
+
+    exp_time = stoptime - starttime - exptime_loss  # exposure time in seconds
+
+    return exp_time, expstart, expstop, good_events
+
+
+def events_to_accum(events_data, size_x, size_y, highres):
+    """Map timetag events to a 2d accum image array.
+
+        Parameters
+        ----------
+        events_data: record array
+            Record array of timetag events.
+
+        size_x: int
+            Number of pixels on axis 1 of the detector.
+
+        size_y: int
+            Number of pixels on axis 2 of the detector.
+
+        highres: bool
+            Boolean value indicating whether the output accum image is in high or low resolution.
+
+        Returns
+        -------
+        accum: array
+            2d image of all events in the imset on the detector.
+
+
+"""
+    # Extract (x,y) event locations from events_table
+    axis1 = events_data['AXIS1']
+    axis2 = events_data['AXIS2']
+
+    # Determine resolution-appropriate binning
+    if highres:
+        range_y = size_y
+        range_x = size_x
+    else:
+        range_y = size_y * 2
+        range_x = size_x * 2
+
+    # Map events to an accum image using a 2d histogram
+    accum, xedges, yedges = np.histogram2d(axis2, axis1, bins=[size_y, size_x], range=[[1, range_y], [1, range_x]])
+
+    return accum
+
+
+if __name__ == "__main__":
+    inttag("gtigap_tag.fits", "test.fits",  increment=200, rcount=6, verbose=True, highres=False, allevents = False)
```

Comment:
I have the same question for you guys here that I did for `tastis`, how runnable from the command line do we want this to be?  You could use `argparse` to enable some argument passing from command line, but maybe that's not worth it since we're not using entry points anyway in `stistools`.

Reply:
Not quite sure on this. It seems like other stistools tools have command line argument passing available, so in the name of keeping things consistent maybe we do want this to be command line runnable? I can iterate with the rest of the team to gauge the importance of this.